File: backend/schemas/branding.py
from pydantic import BaseModel, Field
from typing import Dict, Optional


class BrandingResponse(BaseModel):
    """品牌/表现差异化配置（公开接口，登录页也要用）。

    每个实例从自己的 .env 读取 BRAND_* 配置，前端启动时拉取并应用。
    文字字段为可信 HTML（仅来自后端配置，不接受用户输入）。
    """
    brand_sidebar: str = Field(..., description="侧边栏品牌名（HTML）")
    brand_login: str = Field(..., description="登录页品牌名（HTML）")
    brand_mark: str = Field(..., description="登录页方块标记字符")
    page_title: str = Field(..., description="浏览器标签标题")
    login_headline: str = Field(..., description="登录页主标语（HTML）")
    login_sub: str = Field(..., description="登录页副标语")
    org_scope: str = Field("全公司", description="首页副标题范围词（如 全公司 / 营销销售部）")
    dept_unit: str = Field("部门", description="首页副标题部门单位词（如 部门 / 团队）")
    logo_url: str = Field("", description="侧边栏 logo 图地址（空则前端用内置默认）")
    favicon_url: str = Field("", description="favicon 地址（空则用默认）")
    theme: Dict[str, str] = Field(default_factory=dict, description="CSS 变量覆盖（仅含非空项）")
    sales_code_enabled: bool = Field(False, description="是否启用内部销售码管理（部署级；控制菜单可见）")


class BrandingFull(BaseModel):
    """品牌完整配置（管理员读/写，颜色为扁平字段）。"""
    brand_sidebar: str = ""
    brand_login: str = ""
    brand_mark: str = ""
    page_title: str = ""
    login_headline: str = ""
    login_sub: str = ""
    org_scope: str = ""
    dept_unit: str = ""
    logo_url: str = ""
    favicon_url: str = ""
    accent: str = ""
    accent_hover: str = ""
    accent_soft: str = ""
    sidebar_bg: str = ""
    sidebar_hover: str = ""


class BrandingUpdate(BaseModel):
    """品牌更新（仅传入的字段会被写入；None 跳过、空串=清空回退 .env）。"""
    brand_sidebar: Optional[str] = None
    brand_login: Optional[str] = None
    brand_mark: Optional[str] = None
    page_title: Optional[str] = None
    login_headline: Optional[str] = None
    login_sub: Optional[str] = None
    org_scope: Optional[str] = None
    dept_unit: Optional[str] = None
    logo_url: Optional[str] = None
    favicon_url: Optional[str] = None
    accent: Optional[str] = None
    accent_hover: Optional[str] = None
    accent_soft: Optional[str] = None
    sidebar_bg: Optional[str] = None
    sidebar_hover: Optional[str] = None

File: frontend/src/stores/branding.ts
import { defineStore } from 'pinia'
import { ref, computed } from 'vue'
import defaultLogo from '@/assets/logo.png'

/* 品牌/表现差异化配置（与后端 /api/v1/branding 对应） */
export interface Branding {
  brand_sidebar: string
  brand_login: string
  brand_mark: string
  page_title: string
  login_headline: string
  login_sub: string
  org_scope: string
  dept_unit: string
  logo_url: string
  favicon_url: string
  theme: Record<string, string>
  sales_code_enabled: boolean
}

/* 内置默认值：接口失败或字段缺失时回退（与后端 config.py 默认一致） */
const DEFAULTS: Branding = {
  brand_sidebar: '<span class="brand-accent">P</span>roject <span class="brand-accent">M</span>anager',
  brand_login: '飞书<span class="accent">PM</span>',
  brand_mark: '飞',
  page_title: '飞书项目管理系统',
  login_headline: '把中长期项目<br />管理得<span class="accent">轻盈</span>。',
  login_sub: '里程碑 · 任务 · 风险 · 飞书联动 —— 一处掌握全局。',
  org_scope: '全公司',
  dept_unit: '部门',
  logo_url: '',
  favicon_url: '',
  theme: {},
  sales_code_enabled: false,
}

export const useBrandingStore = defineStore('branding', () => {
  const data = ref<Branding>({ ...DEFAULTS })

  /* 侧边栏 logo：接口给了 URL 用之，否则用构建内置默认图 */
  const logoSrc = computed(() => data.value.logo_url || defaultLogo)

  /* 拉取本实例品牌配置（公开接口，无需 token）。带超时回退默认，避免接口故障导致白屏。 */
  async function fetchBranding() {
    try {
      const ctrl = new AbortController()
      const timer = setTimeout(() => ctrl.abort(), 800)
      const resp = await fetch('/api/v1/branding', { signal: ctrl.signal })
      clearTimeout(timer)
      if (resp.ok) {
        const json = await resp.json()
        data.value = { ...DEFAULTS, ...json }
      }
    } catch {
      // 失败保持默认值（产品默认品牌），不阻断启动
    }
  }

  /* 应用品牌：标签标题 + favicon + 主题色覆盖（在 mount 前调用，避免闪烁） */
  function applyBranding() {
    document.title = data.value.page_title
    if (data.value.favicon_url) {
      let link = document.querySelector<HTMLLinkElement>('link[rel="icon"]')
      if (!link) {
        link = document.createElement('link')
        link.rel = 'icon'
        document.head.appendChild(link)
      }
      link.href = data.value.favicon_url
    }
    // 主题色：覆盖 :root CSS 变量（仅接口返回的非空项；状态语义色不在其列，沿用默认）
    for (const [key, value] of Object.entries(data.value.theme || {})) {
      document.documentElement.style.setProperty(key, value)
    }
  }

  return { data, logoSrc, fetchBranding, applyBranding }
})
